kafka-connect/kafka-connect-runtime/NOTICE

```diff
@@ -0,0 +1,1723 @@
+
+Apache Iceberg
+Copyright 2017-2024 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+--------------------------------------------------------------------------------
+
+This project includes code from Kite, developed at Cloudera, Inc. with
+the following copyright notice:
+
+| Copyright 2013 Cloudera Inc.
+|
+| Licensed under the Apache License, Version 2.0 (the "License");
+| you may not use this file except in compliance with the License.
+| You may obtain a copy of the License at
+|
+|   http://www.apache.org/licenses/LICENSE-2.0
+|
+| Unless required by applicable law or agreed to in writing, software
+| distributed under the License is distributed on an "AS IS" BASIS,
+| WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+| See the License for the specific language governing permissions and
+| limitations under the License.
+
+--------------------------------------------------------------------------------
+
+This binary artifact contains code from the following projects:
+
+--------------------------------------------------------------------------------
+
+Group: org.apache.commons  Name: commons-math3  Version: 3.1.1
+
+Notice: Apache Commons Math
+Copyright 2001-2012 The Apache Software Foundation
+
+This product includes software developed by
+The Apache Software Foundation (http://www.apache.org/).
+
+===============================================================================
+
+The BracketFinder (package org.apache.commons.math3.optimization.univariate)
```

Comment:
I don't see any direct dependency on this. 

I am not even sure this is a right way to generate notice and license. 
Because it looks so much different from spark runtime for example. 

Maybe we need to add how to generate notice and license for modules in "contribute.md" first (also need to mention when do we need to add it, like only for runtime jar modules) and then maybe follow that process.

Reply:
This includes licenses for all transitive dependencies, which I think is correct given those are packaged in the archive. This is the same tool used to generate the license files for the AWS, GCP, and Azure bundles. The Spark runtime predates those. I was thinking we could use this tool for the other runtimes as well.